src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -1942,6 +1942,33 @@ module T = struct
                     (List.map ~f:List.rev detailed) )
               ] ;
         { Staged_ledger_diff.With_valid_signatures_and_proofs.diff })
+
+  let accounts_created t =
```

Comment:
Done.

I think this may not be completely fool-proof: you could have the identical internal command twice, where it creates the account in an earlier block, and we'd misattribute the creation to the latest block (not a problem for user commands, which have nonces). That's unlikely, of course.

Reply:
That's a good point. Can we also check that the state_hash in the leaf (`fst state_hash`) is the same as previous state hash of the block being passed?

Reply:
I'll look at that

Reply:
Done.